projects/fsastorefrontlib/src/cms-components/my-account/inbox/inbox-tab/inbox-messages/inbox-messages.component.ts

```diff
@@ -133,11 +137,16 @@ export class InboxMessagesComponent implements OnInit, OnDestroy {
     if (!checked) {
       this.mainCheckboxChecked = false;
     }
+    let checkedMassagesCounter = 0;
     this.loadedMessages.forEach(message => {
       if (message.uid === messageUid) {
         message.checked = checked;
       }
+      message.checked ? checkedMassagesCounter++ : (checkedMassagesCounter = 0);
```

Comment:
Each time you click on message checkbox, local "checkedMassagesCounter" flag will be set to 0. Then you assign it value 1 if message is checked, which creates new flag identical to message.checked flag ?

This flag is then used to to check if "auto" cursor should be rendered on general checkbox, but in case you have multiple messages checked, but last was changed from checked to unchecked, this logic will not work as intended - you won't be able to check main checkbox?

Reply:
checkedMassagesCounter counts how many messages from table are selected and resets every time when user click on the checkbox. This all happening in the forEach loop.

I fixed this as we agreed instead of ternary operator I'll use if condition to set checkedMassagesCounter.